cli/command/image/build.go

```diff
@@ -247,6 +253,66 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		buildCtx = replaceDockerfileTarWrapper(ctx, buildCtx, relDockerfile, translator, &resolvedTags)
 	}
 
+	// write ctx tar to disk, to make it readable for Image Reference extraction
+	originCtx := buildCtx
+	defer originCtx.Close()
+	readbackTar, err := ioutil.TempFile("", "docker-ctx")
```

Comment:
To be more specific, unless I'm missing something, we could make the call to `replaceDockerfileTarWrapper` above happen even for non-trusted builds, and avoid rewriting any references in the non-trusted case. The `translator` function would pull out registry references. As a cleanup, we could change `replaceDockerfileTarWrapper` into something more generic where the rewriting is optional and is only enabled for trusted builds. It's not clear to me why two passes over the tar are necessary.